docs/metrics.md

```diff
@@ -0,0 +1,161 @@
+  | thanos_cache_redis_hits_total                     | counter   | Total number of items requests to the cache that were a hit.                                                    |
+  | thanos_cache_redis_requests_total                 | counter   | Total number of items requests to redis.                                                                        |
+  | thanos_memcached_client_info                      | gauge     | A metric with a constant '1' value labeled by configuration options from which memcached client was configured. |
+  | thanos_memcached_operations_total                 | counter   | Total number of operations against memcached.                                                                   |
+  | thanos_memcached_operation_failures_total         | counter   | Total number of operations against memcached that failed.                                                       |
+  | thanos_memcached_operation_skipped_total          | counter   | Total number of operations against memcached that have been skipped.                                            |
+  | thanos_memcached_operation_duration_seconds       | histogram | Duration of operations against memcached.                                                                       |
+  | thanos_memcached_operation_data_size_bytes        | histogram | Tracks the size of the data stored in and fetched from memcached.                                               |
+  | thanos_redis_operation_duration_seconds           | histogram | Duration of operations against redis.                                                                           |
+
+### List of Metrics Exported By Thanos Sidecar Shipper
+- When the Thanos Sidecar is run with the `--shipper.upload-compacted` flag, it will sync all older existing blocks from Prometheus local storage on startup.
+
+  | Metric Name                            | Type    | Description                                                              |
+  |----------------------------------------|---------|--------------------------------------------------------------------------|
+  | thanos_shipper_upload_compacted_done   | gauge   | If 1 it means shipper uploaded all compacted blocks from the filesystem. |
+  | thanos_shipper_dir_sync_failures_total | counter | Total number of failed dir syncs.                                        |
+  | thanos_shipper_dir_syncs_total         | counter | Total number of dir syncs.                                               |
+  | thanos_shipper_upload_failures_total   | counter | Total number of block upload failures.                                   |
+  | thanos_shipper_uploads_total           | counter | Total number of uploaded blocks.                                         |
+
+### List of Metrics Exported By Thanos Verifier
```

Comment:
The comment I made in the Thanos Replicator section also applies here.

Reply:
I figured the `thanos_verifier_*` metric is exported by the Thanos Compactor after reading through the bucket verify tooling functionality, I will adjust it to the Compactor PR accordingly.

Reply:
On a quick check through the code I couldn't find any metric named like `thanos_verifier_*`. Where did you see them? Also the metric you have listed under this heading is not what you say it is, as it's named `thanos_bucket_uiblocks_meta_syncs` (see one line below in the source code). I would also don't think it would be exported by the Compactor. If it's part of bucket verification, then it's part of the Thanos bucket tool.

Reply:
Apologies for `thanos_verifier_*`, i meant `thanos_verify_*, i will write its metric to the `thanos/docs/components/tools,md` file accordingly.